src/SharedWeb/Utilities/ServiceCollectionExtensions.cs

```diff
@@ -716,7 +695,7 @@ public static void AddWebAuthn(this IServiceCollection services, GlobalSettings
         }
         else
         {
-            services.AddKeyedSingleton<IDistributedCache, MemoryDistributedCache>("persistent");
+            services.AddKeyedSingleton("persistent", (s, _) => s.GetRequiredService<IDistributedCache>());
```

Comment:
Use whatever was registered for the non-keyed service instead of registering a new instance.